From 65f0e4b251e933d3101bb3d1eec46445a832ccc1 Mon Sep 17 00:00:00 2001
From: Chen Liangjun <b36089@freescale.com>
Date: Thu, 27 Dec 2012 11:47:51 +0800
Subject: [PATCH 377/505] ENGR00238302 ASRC TEST: fix noise bug for too much
 data from driver

When sending a buffer(to be converted) into driver, user application
would tell driver how much data it need. In the old code, the expected
length is misconfigured. In this case, noise would be heard in some
sample rate pair.

In this patch, fix output length configure error.

Signed-off-by: Chen Liangjun <b36089@freescale.com>
---
 test/mxc_asrc_test/mxc_asrc_test.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/test/mxc_asrc_test/mxc_asrc_test.c b/test/mxc_asrc_test/mxc_asrc_test.c
index 4f54017..f18bdaf 100644
--- a/test/mxc_asrc_test/mxc_asrc_test.c
+++ b/test/mxc_asrc_test/mxc_asrc_test.c
@@ -192,8 +192,6 @@ int play_file(int fd_asrc, struct audio_info_s *info)
 	    asrc_get_output_buffer_size(DMA_BUF_SIZE, info->sample_rate,
 					info->output_sample_rate);
 
-	tail = info->channel * 4 * 4;
-
 	convert_flag = 1;
 	memset(input_null, 0, DMA_BUF_SIZE);
 	if ((err = ioctl(fd_asrc, ASRC_START_CONV, &pair_index)) < 0)
@@ -214,7 +212,7 @@ int play_file(int fd_asrc, struct audio_info_s *info)
 			buf_info.input_buffer_length = DMA_BUF_SIZE;
 		}
 
-		buf_info.output_buffer_length = output_dma_size + tail;
+		buf_info.output_buffer_length = output_dma_size;
 		buf_info.output_buffer_vaddr = output_p;
 		if ((err = ioctl(fd_asrc, ASRC_CONVERT, &buf_info)) < 0)
 			goto error;
-- 
1.8.0

